adlfs/spec.py

```diff
@@ -268,7 +268,7 @@ def __init__(
         self.client_id = client_id or os.getenv("AZURE_STORAGE_CLIENT_ID")
         self.client_secret = client_secret or os.getenv("AZURE_STORAGE_CLIENT_SECRET")
         self.tenant_id = tenant_id or os.getenv("AZURE_STORAGE_TENANT_ID")
-        self.anon = anon
+        self.anon = anon or os.getenv("AZURE_STORAGE_ANON", True)
```

Comment:
What kind of values are you expecting here? I think you'll need to cast from whatever `os.getenv("AZURE_STORAGE_ANON")` is to some kind of bool. I think we want something like `"0"` to be `False`. And maybe `"false"` to be `False` as well.

We'll need to document this.

Reply:
Sorry, I had it backwards :/ Your proposal looks good.